dkg/protocol.go

```diff
@@ -10,6 +11,54 @@ type KeyGenOutput struct {
 	Share           *bls.SecretKey
 	OperatorPubKeys map[types.OperatorID]*bls.PublicKey
 	ValidatorPK     types.ValidatorPK
+	Threshold       uint64
+}
+
+type keygenOutputJSON struct {
+	Share           []byte
+	OperatorPubKeys map[types.OperatorID][]byte
+	ValidatorPK     types.ValidatorPK
+	Threshold       uint64
+}
+
+func (k *KeyGenOutput) MarshalJSON() ([]byte, error) {
```

Comment:
We use the Encoder interface not marshalJSON
https://github.com/bloxapp/ssv-spec/blob/main/types/encoder.go